####################
## Culling settings
####################
pcd_input_file: "env2.pcd"
voxel_res: 0.1
frame_id: "world"
debug_enabled: true

####################
## Sensor Position - In reference to body frame - for Culling
####################
sensor_roll: 0
sensor_pitch: 0.093
sensor_yaw: 0
sensor_x: 0
sensor_y: 0
sensor_z: -0.055
sensor_hor_fov: 60
sensor_ver_fov: 45
sensor_near_limit: 0.5
sensor_far_limit: 5.0

############
## Octomap Manager settings
############
tf_frame: "world"
robot_frame: "base_point_cloud"
resolution: 0.1
mesh_resolution: 1.0
visualize_max_z: 5.0
sensor_max_range: 5.0
map_publish_frequency: 0.08
probability_hit: 0.7
probability_miss: 0.4
threshold_min: 0.12
threshold_max: 0.97
threshold_occupancy: 0.7
treat_unknown_as_occupied: false
latch_topics: false
max_free_space: 0
min_height_free_space: 0
map_publish_frequency: 1
treat_unknown_as_occupied: true
change_detection_enabled: false

#visualize_min_z(-std::numeric_limits<double>::max())
#visualize_max_z(std::numeric_limits<double>::max())

# What are these ?
#    nh_private_.param("label/object_certainty_threshold", label_params.obj_certainty_threshold, label_params.obj_certainty_threshold);
#    nh_private_.param("label/object_interest_value", label_params.interest_value, label_params.interest_value);
#    nh_private_.param("label/object_timestamp", label_params.timestamp, label_params.timestamp);
#   <arg name="pcl_topic" default="/pointcloud" />
#   <arg name="stl_file_path" default="" />
  
# logging param
/exploration_node/log/on: true

# initial position param
#exploration_node/init/pose/x: -5
#/exploration_node/init/pose/y: 39
#/exploration_node/init/pose/z: 1.209
#/exploration_node/init/pose/yaw: 3.14

/exploration_node/init/pose/x: 1
/exploration_node/init/pose/y: 2
/exploration_node/init/pose/z: 1
/exploration_node/init/pose/yaw: -3.032297

#4 2 2 0 
# Camera Params
/exploration_node/camera/pitch: [15.0]
/exploration_node/camera/horizontal: [58.0]
/exploration_node/camera/vertical: [45.0]

# Robot Params
/exploration_node/robot/bbx/x: 0.3
/exploration_node/robot/bbx/y: 0.3
/exploration_node/robot/bbx/z: 0.3
/exploration_node/robot/bbx/overshoot: 0.5

# Exploration Algorithm Params 
# - ViewpointGeneration
## Geometric Generation
/exploration_node/exp/gen/res_x: 0.5
/exploration_node/exp/gen/res_y: 0.5
/exploration_node/exp/gen/res_z: 0.5
#/exploration_node/exp/gen/res_yaw: 0.785


# - viewpointEvaluation
/exploration_node/exp/gain/range: 1

#Termination
/exploration_node/exp/ter/num_iteration: 100


#StepSize 
/exploration_node/exp/ter/step_size: 1

# Environment Related Paramters 
# bounding box: necessary to limit the simulation 
# scenario (smaller than actual gazebo scenario)

# Exploration Area (Stanford dataset for area1 and area6)
#/exploration_node/env/bbx/minX: 1.0
#/exploration_node/env/bbx/maxX: 1.5
#/exploration_node/env/bbx/minY: -2.5
#/exploration_node/env/bbx/maxY: 40
#/exploration_node/env/bbx/minZ: 0.0
#/exploration_node/env/bbx/maxZ: 3.0

# Exploration Area (Stanford dataset for area3)
#/exploration_node/env/bbx/minX: 0.0
#/exploration_node/env/bbx/maxX: 350
#/exploration_node/env/bbx/minY: -12
#/exploration_node/env/bbx/maxY: 10
#/exploration_node/env/bbx/minZ: 0.0
#/exploration_node/env/bbx/maxZ: 3.0

# Exploration Area for bosch dataset
#/exploration_node/env/bbx/minX: -2.0
#/exploration_node/env/bbx/maxX: 7
#/exploration_node/env/bbx/minY: -4.5
#/exploration_node/env/bbx/maxY: 6
#/exploration_node/env/bbx/minZ: 0.0
#/exploration_node/env/bbx/maxZ: 2.0
#/exploration_node/env/bbx/softBounds: false

# Exploration Area for the manual Generated Area (CAD)
#/exploration_node/env/bbx/minX: -5.2
#/exploration_node/env/bbx/maxX: 5.2
#/exploration_node/env/bbx/minY: -4
#/exploration_node/env/bbx/maxY: 4
#/exploration_node/env/bbx/minZ: 0.0
#/exploration_node/env/bbx/maxZ: 3.5
#/exploration_node/env/bbx/softBounds: false

# NewArea:  Exploration Area for the manual Generated Area (CAD)
/exploration_node/env/bbx/minX: -6.0
/exploration_node/env/bbx/maxX:  6.0
/exploration_node/env/bbx/minY: -3
/exploration_node/env/bbx/maxY: 5.2
/exploration_node/env/bbx/minZ: 0.0
/exploration_node/env/bbx/maxZ: 3.0
/exploration_node/env/bbx/softBounds: false

#Mavros related settings
/uav_1/mavros/local_position/frame_id: world           #(string, default: fcu) frame_id for message.
/uav_1/mavros/local_position/tf/send: true             #(bool, default: true)
/uav_1/mavros/local_position/tf/frame_id: world        #(string, default: local_origin) Origin frame_id for TF.
/uav_1/mavros/local_position/tf/child_frame_id: fcu    #(string, default: fcu) Child frame_id for TF.
